auth/auth_test.go

```diff
@@ -30,166 +30,220 @@ import (
 	"golang.org/x/oauth2/google"
 	"google.golang.org/api/option"
 	"google.golang.org/api/transport"
-	"google.golang.org/appengine"
-	"google.golang.org/appengine/aetest"
 )
 
 var (
-	client              *Client
-	ctx                 context.Context
-	testIDToken         string
 	testGetUserResponse []byte
-)
+	testIDToken         string
+	testSigner          cryptoSigner
 
-var defaultTestOpts = []option.ClientOption{
-	option.WithCredentialsFile("../testdata/service_account.json"),
-}
-var testClock = &mockClock{now: time.Now()}
+	optsWithServiceAcct = []option.ClientOption{
+		option.WithCredentialsFile("../testdata/service_account.json"),
+	}
+	optsWithTokenSource = []option.ClientOption{
+		option.WithTokenSource(&internal.MockTokenSource{
+			AccessToken: "test.token",
+		}),
+	}
+	testClock     = &internal.MockClock{Timestamp: time.Now()}
+	testKeySource = &fileKeySource{FilePath: "../testdata/public_certs.json"}
+	testProjectID = "mock-project-id"
+)
 
 func TestMain(m *testing.M) {
-	var (
-		err   error
-		ks    keySource
-		creds *google.DefaultCredentials
-		opts  []option.ClientOption
-	)
-	if appengine.IsDevAppServer() {
-		aectx, aedone, err := aetest.NewContext()
-		if err != nil {
-			log.Fatalln(err)
-		}
-		ctx = aectx
-		defer aedone()
-
-		ks, err = newAEKeySource(ctx)
-		if err != nil {
-			log.Fatalln(err)
-		}
-	} else {
-		ctx = context.Background()
-		opts = defaultTestOpts
-		creds, err = transport.Creds(ctx, opts...)
-		if err != nil {
-			log.Fatalln(err)
-		}
-
-		ks = &fileKeySource{FilePath: "../testdata/public_certs.json"}
+	creds, err := transport.Creds(context.Background(), optsWithServiceAcct...)
+	if err != nil {
+		log.Fatalln(err)
 	}
 
-	client, err = NewClient(ctx, &internal.AuthConfig{
-		Creds:     creds,
-		Opts:      opts,
-		ProjectID: "mock-project-id",
-	})
+	testSigner, err = signerFromCreds(creds.JSON)
 	if err != nil {
 		log.Fatalln(err)
 	}
-	client.keySource = ks
-	client.clock = testClock
+	testIDToken = getIDToken(nil)
 
 	testGetUserResponse, err = ioutil.ReadFile("../testdata/get_user.json")
 	if err != nil {
 		log.Fatalln(err)
 	}
-
-	testIDToken = getIDToken(nil)
 	os.Exit(m.Run())
 }
 
-func TestNewClientServiceAccountSigner(t *testing.T) {
+func TestNewClientWithServiceAccountCredentials(t *testing.T) {
+	creds, err := transport.Creds(context.Background(), optsWithServiceAcct...)
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
Maybe t.Fatal(err) would be preferable? I'm not sure that it makes any difference, though...

Reply:
Good point. I think we should be using `t.Fatal()` inside test cases.